packages/block-editor/src/components/block-list/index.native.js

```diff
@@ -262,8 +262,6 @@ export class BlockList extends Component {
 						{ flex: isRootList ? 1 : 0 },
 						! isRootList && styles.overflowVisible,
 					] }
-					horizontal={ horizontal }
-					numColumns={ 1 }
```

Comment:
After dynamically setting `horizontal` within `packages/block-library/src/columns/edit.native.js`, `numColumns` began to work. However, after debugging this further, `horizontal` appears to be unnecessary. Removing it altogether also negates the need to set `numColumns`. 

The [other conditional styles](https://github.com/WordPress/gutenberg/blob/1177899bae17eee8db477d6402bf54396bc10b85/packages/block-editor/src/components/block-list/index.native.js#L268) controlled by `horizontal` are still applied resulting in the desired layout for both skinny and wide screens.